src/objectivec/oga_sequences.mm

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import "error_utils.h"
+#import "oga_internal.h"
+#import "ort_genai_objc.h"
+
+@implementation OGASequences {
+  std::unique_ptr<OgaSequences> _sequences;
+}
+
+- (instancetype)initWithCXXPointer:(std::unique_ptr<OgaSequences>)ptr {
+  _sequences = std::move(ptr);
+  return self;
+}
+
+- (nullable instancetype)initWithError:(NSError**)error {
+  if ((self = [super init]) == nil) {
+    return nil;
+  }
+
+  try {
+    _sequences = OgaSequences::Create();
+    return self;
+  }
+  OGA_OBJC_API_IMPL_CATCH_RETURNING_NULLABLE(error)
+}
+
+- (size_t)count {
+  return _sequences->Count();
+}
+
+- (nullable OGAInt32Span*)sequenceAtIndex:(size_t)index
+                                    error:(NSError**)error {
+  if (index >= [self count]) {
+    NSDictionary *errorDictionary = @{NSLocalizedDescriptionKey : @"The index is out of bounds"};
+    *error = [[NSError alloc] initWithDomain:kOgaErrorDomain code:-1 userInfo:errorDictionary];
```

Comment:
check `error != nil` first?

to simplify, could just put this in the try block and throw a normal C++ exception.

Reply:
yeah this seems unnecessary, I'll just delete it and make use of the C++ exceptions.